indirect.h

```diff
@@ -45,6 +45,34 @@ namespace xyz {
 }
 #endif  // XYZ_UNREACHABLE_DEFINED
 
+namespace detail {
+template <class T, class U>
+concept three_way_comparable_with = requires(T& t, U& u) {
+  { t <=> u } -> std::convertible_to<std::weak_ordering>;
+};
+
+// See: https://eel.is/c++draft/expos.only.entity
+template <class T, class U>
+constexpr auto synth_three_way(const T& t, const U& u)
+  requires requires {
+    { t < u } -> std::convertible_to<bool>;
+    { u < t } -> std::convertible_to<bool>;
+  }
+{
+  if constexpr (three_way_comparable_with<T, U>) {
+    return t <=> u;
+  } else {
+    if (t < u) return std::weak_ordering::less;
```

Comment:
I think the else branch can just be `std:: compare_strong_order_fallback`: https://en.cppreference.com/w/cpp/utility/compare/compare_strong_order_fallback

Reply:
std::compare_weak_order_fallback requires that they are the same type and we need to support comparison between different but comparable types.

Reply:
Good point, thanks for the clarifying.